front-end/src/reservations/edit-new/EditReservation.js

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getReservationByReservationId } from "../../utils/api";
+import ReservationForm from "./ReservationForm";
+
+function EditReservation({ setReservations }) {
+  const { reservation_id } = useParams();
+  const [editReservation, setEditReservation] = useState({});
+
+  useEffect(getReservation, []);
+
+  async function getReservation() {
```

Comment:
might just be personal preference, but i would just do this inside the useeffect directly. This is really small to be extracted out and would look cleaner just as an anonymous function inside the useEffect. Just IMO though